test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -1129,11 +1132,33 @@ public async Task GetTrustResultAsync_WithAlwaysVerifyCountersignatureBehavior_R
                     Assert.Equal("Verification settings require a repository countersignature, but the package does not have a repository countersignature.", error.Message);
                 }
             }
+
+            [CIOnlyFact]
+            public async Task GetTrustResultAsync_WithExpiredSignature_ReturnsValidAsync()
+            {
+                using (X509Certificate2 certificate = await GetExpiringCertificateAsync(_fixture))
+                using (Test test = await Test.CreateRepositoryPrimarySignedPackageAsync(certificate))
+                using (var packageReader = new PackageArchiveReader(test.PackageFile.FullName))
+                {
+                    await SignatureTestUtility.WaitForCertificateExpirationAsync(certificate);
+
+                    PrimarySignature primarySignature = await packageReader.GetPrimarySignatureAsync(CancellationToken.None);
+                    PackageVerificationResult status = await _provider.GetTrustResultAsync(packageReader, primarySignature, _defaultSettings, CancellationToken.None);
+
+                    Assert.Equal(SignatureVerificationStatus.Valid, status.Trust);
```

Comment:
If I understood correctly it's not testing timestamp. But do we still test repository certificate now?

Reply:
Yes.  This test verifies that in the default accept mode an expired repository primary signature raises warnings but the package still is trusted as unsigned.  That is because accept mode allows both unsigned and signed packages, and a package with an expired signature is considered unsigned.